smartsim/_core/launcher/dragon/dragonConnector.py

```diff
@@ -0,0 +1,447 @@
+                    stdout=dragon_out.fileno(),
+                    cwd=path,
+                    shell=False,
+                    env=current_env,
+                    start_new_session=True,
+                )
+
+            if connector_socket is None:
+                raise SmartSimError("Socket failed to initialize")
+
+            def log_dragon_outputs() -> None:
+                if self._dragon_head_process:
+                    self._dragon_head_process.wait(1.0)
+                    if self._dragon_head_process.stdout:
+                        for line in iter(
+                            self._dragon_head_process.stdout.readline, b""
+                        ):
+                            logger.info(line.decode("utf-8").rstrip())
+                    if self._dragon_head_process.stderr:
+                        for line in iter(
+                            self._dragon_head_process.stderr.readline, b""
+                        ):
+                            logger.warning(line.decode("utf-8").rstrip())
+                    logger.warning(self._dragon_head_process.returncode)
+
+            if address is not None:
+                server = dragonSockets.as_server(connector_socket)
+                logger.debug(f"Listening to {socket_addr}")
+                request = _assert_schema_type(server.recv(), DragonBootstrapRequest)
+
+                logger.debug(f"Connecting to {request.address}")
+                server.send(
+                    DragonBootstrapResponse(dragon_pid=self._dragon_head_process.pid)
+                )
+
+                connector_socket.close()
+                self._set_timeout(self._timeout)
+                self._handshake(request.address)
+
+                # Only the Connector which started the server is
+                # responsible of it, that's why we register the
+                # cleanup in this code branch.
+                # The cleanup function should not have references
+                # to this object to avoid Garbage Collector lockup
+                server_socket = self._dragon_head_socket
+                server_process_pid = self._dragon_head_process.pid
+
+                if server_socket is not None and self._dragon_head_process is not None:
+                    atexit.register(
+                        _dragon_cleanup,
+                        server_socket=server_socket,
+                        server_process_pid=server_process_pid,
+                        server_authenticator=self._authenticator,
+                        graceful=self._graceful_cleanup,
+                    )
+            else:
+                # TODO parse output file
+                log_dragon_outputs()
+                raise SmartSimError("Could not receive address of Dragon head process")
+
+    def cleanup(self, graceful: bool = True) -> None:
+        if self._dragon_head_socket is not None and self._dragon_head_pid is not None:
+            _dragon_cleanup(
```

Comment:
I feel like we should be able to send the shutdown request and then just go away, now that you have improved the shutdown in the backend?

Reply:
Yeah, let's see if that flies.